File: rookie-ui/src/types/api/knowhub/tag.ts
/**
 * 文件作用：
 * 定义博客受控标签管理对接后端所需的接口类型，
 * 与后端 TagVo 对齐，供 api 层和页面层统一消费。
 * 关键约定：
 * - 标签列表**不分页**，后端返回 List<TagVo>，前端用 ApiResult<TagRecord[]> 消费。
 * - status 为 int（0 禁用 / 1 启用），与字典数据型的数值口径一致，不走字符串字典。
 */
import type { PageQueryParams } from '../system/common'

/**
 * 受控标签记录，与后端 TagVo 字段对齐。
 */
export interface TagRecord {
  tagId?: number
  tagName: string
  description?: string
  sort?: number
  status: number
  createBy?: string
  createTime?: string
  updateBy?: string
  updateTime?: string
}

/**
 * 标签列表查询参数，与后端 TagVo（作为查询入参）对齐。
 * 后端 /tag/list 不分页，pageNum/pageSize 仅占位以兼容 PageQueryParams 约定，实际不传后端。
 */
export interface TagListQuery extends Partial<PageQueryParams> {
  tagName?: string
  status?: number
}

/**
 * 标签新增/编辑入参，与后端 TagVo 对齐。
 * 编辑时 tagId 必填。
 */
export type TagPayload = TagRecord

File: knowhub/src/main/java/com/knowhub/pojo/entity/Tag.java
package com.knowhub.pojo.entity;

import com.rookie.common.pojo.BaseEntity;

import java.util.Date;

/**
 * 受控标签实体，对应 tag 表。
 * 仅管理员可维护；博客发文时从中选择。
 */
public class Tag extends BaseEntity {

    private Long tagId;

    private String tagName;

    private String description;

    private Integer sort;

    private Integer status;

    private Integer deleted;

    public Tag() {
    }

    public Tag(Date createTime, Date updateTime, String createBy, String updateBy,
               Long tagId, String tagName, String description, Integer sort, Integer status, Integer deleted) {
        super(createTime, updateTime, createBy, updateBy);
        this.tagId = tagId;
        this.tagName = tagName;
        this.description = description;
        this.sort = sort;
        this.status = status;
        this.deleted = deleted;
    }

    public Long getTagId() {
        return tagId;
    }

    public void setTagId(Long tagId) {
        this.tagId = tagId;
    }

    public String getTagName() {
        return tagName;
    }

    public void setTagName(String tagName) {
        this.tagName = tagName;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Integer getSort() {
        return sort;
    }

    public void setSort(Integer sort) {
        this.sort = sort;
    }

    public Integer getStatus() {
        return status;
    }

    public void setStatus(Integer status) {
        this.status = status;
    }

    public Integer getDeleted() {
        return deleted;
    }

    public void setDeleted(Integer deleted) {
        this.deleted = deleted;
    }

    @Override
    public String toString() {
        return "Tag{" +
                "tagId=" + tagId +
                ", tagName='" + tagName + '\'' +
                ", sort=" + sort +
                ", status=" + status +
                ", deleted=" + deleted +
                ", createTime=" + getCreateTime() +
                ", updateTime=" + getUpdateTime() +
                ", createBy='" + getCreateBy() + '\'' +
                ", updateBy='" + getUpdateBy() + '\'' +
                '}';
    }
}